### ISSUE TITLE
spack install dealii build error (due to gmsh and sundials)

### ISSUE THREAD

**Issue by tyszwh** [dealii/dealii#9774]
Hi, 
I try to install Dealii by " Spack install Dealii".But some problems arising.
1 The GMSH can not be built correctly untile the Oce package has been removed(~Oce).
2 The sundials-3.2.1 build failed with the error message as follows. 
[spack-build-out.txt](https://github.com/dealii/dealii/files/4400762/spack-build-out.txt)

Is the new version of sundials still not be supported?

**Comment by masterleinad** [comment]
What kind of error do you get when building `gmsh` and `Oce`?

Yes, so far no one has spent the effort to rewrite the sundials interface to also support newer versions. See https://github.com/dealii/dealii/issues/7597.

**Comment by tyszwh** [comment]
The error message from gmsh+oce
[spack-build-out.txt](https://github.com/dealii/dealii/files/4403234/spack-build-out.txt)

So, I need to add the "~sundials"  or change the package.py file for now when I use "spack install dealii". Is it right?

**Comment by jppelteret** [comment]
Hmm... I'm surprised about this, because I've recently managed to build deal.II through Spack on both Linux and OSX with almost all the bells and whistles, and I didn't run into these problems.

I don't know what the problem with GMSH+OCE could be, but you could just specify `spack install dealii ^gmsh~oce` to at least remove that dependence (if you don't need it, of course). As for Sundials, here is the saliant part of the error message
```
-- Found PythonInterp: /usr/bin/python  
CMake Error at config/SundialsAddTest.cmake:30 (IF):
  if given arguments:

    "LESS" "3"

  Unknown arguments specified
```
which, according to the source code, checks
```
  find_package(PythonInterp)
  IF(${PYTHON_VERSION_MAJOR} LESS 3)
    IF(${PYTHON_VERSION_MINOR} LESS 7)
      PRINT_WARNING("Python version must be 2.7.x or greater to run regression tests"
                    "Examples will build but 'make test' will fail.")
    ENDIF()
  ENDIF()
```
So its actually complaining that you need to use a newer version of python in order to install sundials.

**Comment by tyszwh** [comment]
The first problem is solved by removing the conflict opencascade library that exists in another place.
 For Sundials, yes, the error message looks like a syntax error of CMake. The CMake version is  3.14.3, but I can build it correctly through manual configuration (without spack). However, I still cannot figure out the reason.

**Comment by tyszwh** [comment]
This problem is solved when I change the oringal gcc-7.5 to gcc-9.2. It may not related to the gcc version. It should be some conflict in system.
